rasa_core/domain.py

```diff
@@ -416,7 +416,27 @@ def load(cls, filename, action_factory=None):
         )
 
     @classmethod
-    def validate_domain_yaml(cls, filename):
+    def load_from_yaml(cls, yaml, action_factory=None):
+        cls.validate_domain_yaml(yaml, string_input=True)
+        data = read_yaml_string(yaml)
+        utter_templates = cls.collect_templates(data.get("templates", {}))
+        if not action_factory:
+            action_factory = data.get("action_factory", None)
+        slots = cls.collect_slots(data.get("slots", {}))
+        additional_arguments = data.get("config", {})
+        return TemplateDomain(
```

Comment:
why do we have both `cls` and `TemplateDomain` in this method? surely they are the same? if we are allowing for subclassing then probably stick with `cls` ?

Reply:
oh yes, good call